examples/neighbors/plot_nca_dim_reduction.py

```diff
@@ -0,0 +1,101 @@
+"""
+==============================================================
+Dimensionality Reduction with Neighborhood Components Analysis
+==============================================================
+
+Sample usage of Neighborhood Components Analysis for dimensionality reduction.
+
+This example compares different (linear) dimensionality reduction methods
+applied on the Digits data set. The data set contains images of digits from
+0 to 9 with approximately 180 samples of each class. Each image is of
+dimension 8x8 = 64, and is reduced to a two-dimensional data point.
+
+Principal Component Analysis (PCA) applied to this data identifies the
+combination of attributes (principal components, or directions in the
+feature space) that account for the most variance in the data. Here we
+plot the different samples on the 2 first principal components.
+
+Linear Discriminant Analysis (LDA) tries to identify attributes that
+account for the most variance *between classes*. In particular,
+LDA, in contrast to PCA, is a supervised method, using known class labels.
+
+Neighborhood Components Analysis (NCA) tries to find a feature space such
+that a stochastic nearest neighbor algorithm will give the best accuracy.
+Like LDA, it is a supervised method.
+
+One can see that NCA enforces a clustering of the data that is visually
+meaningful even after the large dimensionality reduction.
```

Comment:
despite the large reduction in dimension.

Reply:
That's better indeed